### ISSUE TITLE
Bug: [member-ordering] Alphabetically sort does not pay attention to the use of fields

### ISSUE THREAD

**Issue by sviat9440** [typescript-eslint/typescript-eslint#5862]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=MYGwhgzhAECC0G8CwAoa0BG0C80CMA3KumDtAC4AWAlhAHQZEoC+qQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0YANqQ+0IZAA0mSKIBmAQwT4pwSENHQpkTfGIALTb3yUyZ+IUgBfZwF1wYZyGdA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class A {
  b = 1;
  a = this.b;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/member-ordering": ["error", {"default": {"order": "alphabetically"}}]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error, because it is impossible to arrange the fields differently

### Actual Result

Member a should be declared before member b

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.40.1` |
| `@typescript-eslint/parser` | `5.40.1` |
| `TypeScript` | `4.5.2` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2882

Please use the search
